cmd/ingester/app/consumer/sepukku.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consumer
+
+import (
+	"runtime"
+	"strconv"
+	"sync/atomic"
+	"time"
+
+	"github.com/uber/jaeger-lib/metrics"
+	"go.uber.org/zap"
+)
+
+// seppukuFactory is a factory for seppukuWorkers
+type seppukuFactory struct {
+	metricsFactory metrics.Factory
+	logger         *zap.Logger
+	interval       time.Duration
+	panicFunc      func(int32)
+}
+
+type seppukuWorker struct {
+	msgConsumed    *uint64
+	ticker         *time.Ticker
+	closePartition chan struct{}
+}
+
+func newSeppukuFactory(factory metrics.Factory, logger *zap.Logger, interval time.Duration) seppukuFactory {
+	return seppukuFactory{
+		metricsFactory: factory,
+		logger:         logger,
+		interval:       interval,
+		panicFunc: func(partition int32) {
+			factory.Counter("seppuku", map[string]string{"partition": strconv.Itoa(int(partition))}).Inc(1)
+			time.Sleep(time.Second) // Allow time to flush metric
+
+			buf := make([]byte, 1<<20)
+			logger.Panic("No messages processed in the last check interval",
+				zap.Int32("partition", partition),
+				zap.String("stack", string(buf[:runtime.Stack(buf, true)])))
```

Comment:
in hotrod zap logger automatically prints the stack. What's different here?

Reply:
zap only prints the current go routine - This prints out all go routines.